packages/workspace/src/utils/rules/to-js.ts

```diff
@@ -15,9 +28,79 @@ export function toJS(): Rule {
         });
         return {
           content: Buffer.from(result, 'utf-8'),
-          path: normalize(entry.path.replace(/\.tsx?$/, '.js')),
+          path,
         };
       }
     )
+  )(tree, context);
+};
+
+const changeExtensionsInConfigFiles: Rule = (tree, context) => {
+  const files = new Set(
+    tree.actions
+      .filter((action) => action.kind === 'c')
```

Comment:
At one point, did you have it so that the kind could be `overwrite`? I see a lot of removed code that seems to assume that it might be taken care of here?

Reply:
Yeah, you nailed it. That's exactly what happened.

I think I would have ot handle `overwrite` differently, only change new lines, not existing lines.